for-upstream-only/nixos/modules/services/continuous-integration/hercules-ci-agent.nix

```diff
@@ -68,11 +70,11 @@ in
       };
     };
 
-    # FIXME: Is this the right approach?
-    users.extraUsers.hercules-agent =
-      if config.ids.uids ? "hercules-agent"
-      then { uid = config.ids.uids.hercules-agent; } // agentUser
-      else agentUser;
-
+    users = mkIf (cfg.user == defaultUser) {
+      users.hercules-agent =
+        if config.ids.uids ? "hercules-ci-agent"
```

Comment:
I don't think we need the uid anyway, but I guess it doesn't hurt to have this here :)